env_vars.tf

```
@@ -44,7 +44,8 @@ data "ns_secret_keys" "this" {
 }
 
 locals {
-  secret_keys  = data.ns_secret_keys.this.secret_keys
-  all_secrets  = data.ns_env_variables.this.secrets
-  all_env_vars = data.ns_env_variables.this.env_variables
+  secret_keys          = data.ns_secret_keys.this.secret_keys
+  all_secrets          = data.ns_env_variables.this.secrets
+  all_env_vars         = data.ns_env_variables.this.env_variables
+  existing_secret_refs = [for key in data.ns_env_variables.this.secret_refs : { name = key, valueFrom = data.ns_env_variables.this.secret_refs[key] }]
```

You should be able to simplify this...

```suggestion
  existing_secret_refs = [for key, ref in data.ns_env_variables.this.secret_refs : { name = key, valueFrom = ref }]
```

---

Done